### PR TITLE
Updates dreamcheck and fixes the code so its not shit

### PR DESCRIPTION
**Pull request by alexkar598**
Updates travis to dreamchecker 1.6 and fixes bugs in the codebase that were newly detected.

closes #10722

### CODE REVIEW

**Comment by redmoogle** [comment]
Golf is a incomprehensible buggy mess, also looks shit. I'd either nuke it from the codebase, or fix it

**Review by redmoogle — APPROVED** [review]
Eh fuck it, whos gonna bus golf, but theres gonna need to be fixes later

**Comment by alexkar598** [comment]
golf works fine I dont understand what's your thing with it

**Comment by redmoogle** [comment]
> 
> 
> golf works fine I dont understand what's your thing with it

well since you changed how CanAllowPass or whatever works on it, everytime you walk over a golfhole it will trigger the lands in the hole message, along with anytime a golfball goes flying over

File: code/game/golf/golf.dm [hunk 1]
```diff
@@ -27,11 +27,13 @@
...
 				src.anchored = 0
 
-/obj/machinery/golfhole/CanPass(atom/movable/mover, turf/target, height=0)
-	if (contents.len >= 3)
-		visible_message("<span class='notice'>The golf hole is full! Try removing golfballs from the hole.</span>")
-		return ..(mover, target, height)
+/obj/machinery/golfhole/Cross(atom/movable/mover, turf/target)
+	. = ..()
 	if (istype(mover,/obj/item/golfball) && mover.throwing  && anchored)
+		if (contents.len >= 3)
```
Comment (alexkar598) [inline code/game/golf/golf.dm:33]: Except I moved the check here

Comment (redmoogle) [reply code/game/golf/golf.dm:33]: yeah I get the it landed in the hole despite that not even being true

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -657,12 +657,19 @@ GLOBAL_LIST_EMPTY(teleportlocs)
   * * Gravity if the Z level has an SSMappingTrait for ZTRAIT_GRAVITY
   * * otherwise no gravity
   */
-/atom/proc/has_gravity(turf/T)
-	if(!T || !isturf(T))
+/atom/proc/has_gravity(turf/target)
+	SHOULD_BE_PURE(TRUE)
```
Comment (redmoogle) [inline code/game/area/areas.dm:661]: Unneeded `SHOULD_BE_PURE(TRUE)`

**Review by redmoogle — CHANGES_REQUESTED** [review]

**Review by redmoogle — APPROVED** [review]
'well to be honest it was just to help me debug but I left it there because it doesnt hurt and has gravity should in fact, be pure'

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**